### DISCUSSION TITLE
Support for GitHub

### DISCUSSION THREAD

**Discussion by Dip123s** [community/community#23004] [New to GitHub]
<p>Hello Team,</p>
<p>We often have a situation wherein some of the developers inadvertently push code on his public repos containing some sensitive information. Generally this is followed by frantic attempts to clean this up. This involves deleting the repository and finding the number of views, fetches,forks,clones etc made for this repository. While Github has published its policy on sensitive data removal and information on traffic (which I believe is for some 14 days), I would like to know if Github support can provide this information if the incident gets detected too late viz maybe a month later after the actual upload.</p>

**Comment by ThirdEyeRose** [comment, marked as answer]
<p>Hi <a class="mention" href="/dip123s">@dip123s</a>,</p>
<p>For information about what Support can do in your specific situation, you will want to <a href="https://github.com/contact" rel="nofollow noopener">reach out to GitHub Support directly</a>.</p>
<p>That being said, you say that this is a fairly common scenario for your developers? If so, we might be able to make some recommendations to reduce this problem as well. After all, the best way to stop a data breach is with prevention.</p>
<p>Thanks! </p>

**Comment by Dip123s** [comment]
<p>Thanks. Would appreciate recommendations from you. Pls suggest</p>

**Comment by ThirdEyeRose** [comment]
<p>Hi <a class="mention" href="/dip123s">@dip123s</a>,</p>
<p>Well, since your problem seems to be related to pushing of sensitive data to public repositories, I would recommend setting up a <a href="https://help.github.com/en/articles/about-protected-branches" rel="nofollow noopener">review system using protected branches</a> where only certain people have access to push to those repositories.</p>
<p>If the sensitive data is of a specific type, it is also possible to <a href="https://help.github.com/en/articles/about-status-checks" rel="nofollow noopener">set up status checks</a> which would make sure that each commit meets a certain criteria. <a href="https://help.github.com/en/articles/about-token-scanning" rel="nofollow noopener">GitHub already does a limited amount of token scanning</a> to help with this.</p>
<p>Hope this helps!</p>
